app/models/otheremployment/ShareAwardedShareSchemePlanType.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.otheremployment
+
+import play.api.libs.json.{Format, Json}
+
+object ShareAwardedShareSchemePlanType extends Enumeration {
```

Comment:
Can you use a sealed trait with case objects here?

Reply:
No I don't think so, I think Enumeration in Scala 2.x should almost never be used. I would say we ought to make a tech-debt ticket to stop using Enumeration in the rest of the service.

[Here](https://pedrorijo.com/blog/scala-enums/)'s a good article about Enumeration vs sealed traits. The main problem is if you use Enumeration you can't do exhaustive matching during compilation which is pretty essential!

Reply:
Added tech debt ticket for it